packages/react/index.js

```diff
@@ -1,6 +1,38 @@
 'use strict';
 
+function testMinification() {
+  if (testMinification.name === 'testMinification') {
+    // We are not minified.
+    // This might be a Node environment where DCE is not expected anyway.
+    return;
+  }
+  if (process.env.NODE_ENV === 'development') {
+    // We expect this method only to be called in production.
+    throw new Error('This is unreachable');
+  }
+  try {
+    const source = testMinification.toString();
+    if (source.indexOf('toString') === -1) {
+      // We know for a fact the above line exists.
+      // Therefore the browser gave us invalid source.
+      return;
+    }
+    if (source.indexOf('unreachable') !== -1) {
```

Comment:
This is always true because it will find its own search string, i.e.
```js
(function foo() { return foo.toString().indexOf('any arbitrary string') !== -1; })();
```
so this error will be thrown in correctly minified code.

Extracting it to a variable might work:
```js
var searchString = 'any arbitrary string';
(function foo() { return foo.toString().indexOf(searchString) !== -1; })();
```
but Babili [will inline it][inlining], recreating the original problem.

`'UNREACHABLE'.toLowerCase()` appears to work for now, but probably not in the future, as Babili [currently evaluates](https://github.com/babel/minify/tree/c40dab9760fc57cd876f0d5a1e1cba0b2ac5e9ff/packages/babel-plugin-minify-constant-folding) some other `String.prototype` methods.

[inlining]: https://babeljs.io/repl/#?babili=true&evaluate=true&lineWrap=true&presets=&targets=&browsers=&builtIns=false&debug=false&experimental=true&loose=false&spec=false&code_lz=BQMwrgdgxgLglgewsAlAAgN4FgBQuBuAhgE5oDOApiVABYDKMxcEA5mgLxoDkkxVthAEYAbClwDcuXOGjwkaGBTIwAsszgg4UQnOTpsONGg1pgi5WogatOxBAB0EQgFsKHdpy7nV6zdt1c-rhGRgD0oWgA6m4kbhAIMGjOvnAUACb2wSHhaAAqNHBkSXAsNImCMWgAcghpbhQQ-HDESK4QiQDuNBR8aAAiAMIAosZF8YkUAB4ADhSw6WiEEACeHYTLmYYhfDBgxBCSWwC-WSbAXNMtaWCwdlzunnX4FMII020wgZhZYRHRaFNZrAFAUiq4YDRamgkMJlgoEGgKmhtMJRGljBA0JdajddJsQiCWh00BAKMShsQWsRzvlCqM0Lx-DQhKJAocjCctow4QYCVAkMpyAg9lA3JxvJZrP47PYYAgGExWKh2SEzmRhcRRfZmHVJgB5EDnOUK5gsL4eTgAWgAjEEtgScv8ANbxYkgBCkQhoECEYEQmKCBDPNDCZj1SaFGBkfEEtA5fI9Cju3r-xFEyikFiEYNgIrMIih9HqkUUGMEnZ7A4_NCcglqjVanVTA3ASjUeiMU3oACEHjQNrtsbjET6VHR_LqANDyScujQHWFwnRzODyiY01m6IhOjTS1o1d-iLmhFzbjgiTpcGc0wQZDIcBEbjlaD4vpoGKxV1xMoP5AoMFyK8KGFGBQEgW4kFQb57VjHIhggMg9jPRIWH_Io-BvYhFC3BFsWuCDMVeFgWFNX8jAhIkSTJNAKSpc4ACV-AvdDICsVgPzw795GcWoKAAGkRMBEjqQhx147gyKHKSCQAam4F4r2YWx5GZMYEiPBpFg3UN0nsNBGNEtBIWJZ9-UpOYYFhCSYOk2yQjkrh-QgTQWCQ_SmO9D1PxxAiAC5rLswL7O4MoYGmMhfPCEBBD2YR7H5ZxQlfWBLVmYgQEtU9LX9VKvwIy0YrgJc2V_I4UBVDksiOZEdFoUwKH0TlOVwCUUmlSDyqkHAyuVXAgA